jupyterhub/files/hub/jupyterhub_config.py

```diff
@@ -372,6 +372,11 @@ def camelCaseify(s):
     c.Spawner.debug = True
 
 
+# load seeded secrets
+c.JupyterHub.proxy_auth_token = get_secret_value("JupyterHub.proxy_auth_token")
```

Comment:
To clarify: the change here is that these secrets are no longer optional in the chart, and cannot possibly be left unspecified or null, is that correct? The previous logic allowed these to be unspecified, in which case the existing default behavior would occur, but explicitly setting them to null or empty values is not the same thing.

Reply:
It is correct that we will now always provide a value to these three configuration options, if you search for "assert hack" you will find an assertion I make about that.

Any falsy value for proxy.secretToken, hub.cookieSecret, hub.config.CryptKeeper.keys would lead to automatic generation of a new passwords.

Is this indirectly influencing the behavior of JupyterHub?

Reply:
`get_secret_value("JupyterHub.proxy_auth_token")` returns the hub k8s Secret's key's value, which in turn is set by a named template, which in turn has priority logic to:

1. return any explicitly set value
2. return the previous k8s Secret's value
3. return autogenerated value